### DISCUSSION TITLE
Red border.

### DISCUSSION THREAD

**Discussion by ghost** [david-vanderson/dvui#67] [Q&A] [answered]
I using the version of dvui I cloned several days ago.  I'm rewriting my tabs so that they are easily added and removed from the tab bar and to keep up with the upgrades in the dvui menu and menu items. 

The tab bar items have a red border. I assumed that I need to set options.debug to false and so I did.

1. The red border remains.
2. Using the debug tool, I can make the red border disappear and reappear.

Any thoughts? Thanks again.

![Screenshot from 2024-03-20 14-28-02](https://github.com/david-vanderson/dvui/assets/137411855/3b885644-2377-494e-987f-33e9d8d87a80)

**Comment by iacore** [comment]
If you have the debug window open, the red border shows which widget is hovered over.

**Reply by ghost** [reply]
Yes. My app lets me open the debug window from its main menu.

However, the red border shown in the screen shot happens without me using the debug window. It just happens when the app starts.

Then, when I use the debug window, as I move the mouse away from the tab-bar items, their red border disappears.

**Comment by david-vanderson** [comment]
The red border is controlled by `Window.debug_widget_id`.  It is 0 by default, but perhaps that is colliding with your tab container?  Did you perhaps add some debug code to set it, or force the container to have an id of 0?

If it is purely accidental, then maybe an id of 0 is more likely than I expected.  If that's the case then I'll have to change it to be an optional.

**Reply by ghost** [reply]
Thanks I'll look into that.

**Reply by ghost** [reply]
When I remove the containers it's still there.

**Reply by david-vanderson** [reply]
Can you point me to a version of your project that shows this problem (maybe in a branch)?

**Reply by ghost** [reply]
Thanks, Sure, I wasn't trying to rush you. I just meant that post as an update. I'm thinking it's here in this file. I copied this from you menu items.

Ill get the crud up at github in a minute.

**Reply by ghost** [reply]
I uploaded it to https://github.com/JosephABudd/crud.

The tab menu item file is at https://github.com/JosephABudd/crud/blob/main/src/%40This/deps/widget/TabBarItemWidget.zig

It's containers are in https://github.com/JosephABudd/crud/blob/main/src/%40This/deps/widget/TabBarWidget.zig

**Reply by ghost** [reply]
Hold Everything.

error(dvui): src/@This/frontend/screen/vtab/VTAB/screen.zig:218 id d689e5e9 already used this frame (highlighting), may need to pass .id_extra = <loop index> into Options (see https://github.com/david-vanderson/dvui/blob/master/readme-implementation.md#widget-ids )

**Reply by ghost** [reply]
That error message is for the upload titled "added tab containers" which is the last one I uploaded.

**Reply by ghost** [reply, marked as answer]
It must be that my tab bar item fn init is not adding the id_extra. 

So the red border is a debugging feature you built into dvui. I'm glad I found that out.

**Reply by ghost** [reply]
That was it. Hope I didn't take up any of your time. And I do like that debug feature.

**Reply by iacore** [reply]
I made the error message better.

**Reply by david-vanderson** [reply]
Excellent.  Thanks both of you!
